### PR TITLE
dd: Dependency Descriptor RTP header extension

### PR DESCRIPTION
**Pull request by alfredh**
Defined in:
https://aomediacodec.github.io/av1-rtp-spec/#dependency-descriptor-rtp-header-extension

This work was sponsored by Nvidia Corporation

### CODE REVIEW

File: src/dd/dd.c [hunk 1]
```diff
@@ -0,0 +1,441 @@
...
+	return 0;
+}
+
+
+void dd_print(const struct dd *dd)
+{
+	if (!dd)
+		return;
+
+	printf("~~~~ DD: ~~~~\n");
```
Comment (sreimers) [inline src/dd/dd.c:345]: Is there a specific reason for using `printf` instead of `re_printf` in this function?

File: src/dd/dd.c [hunk 2]
```diff
@@ -0,0 +1,441 @@
...
+		dd->ext = true;
+	}
+	else {
+		no_extended_descriptor_fields(dd);
+	}
+
+#if 0
+	frame_dependency_definition();
+	zero_padding = f(sz * 8 - TotalConsumedBits);
+#endif
```
Comment (sreimers) [inline src/dd/dd.c:334]: Looks like experimental/dead code?

File: src/dd/dd_enc.c [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
+
+	return 0;
+}
+
+
+static int template_fdiffs(struct putbit *pb, const struct dd *dd)
+{
+	int err;
+
+	re_printf("template_fdiffs (encode) bit_pos=%zu\n", pb->bit_pos);
```
Comment (sreimers) [inline src/dd/dd_enc.c:103]: should this `DEBUG_INFO` instead of `re_printf`?

File: src/dd/dd.c (hunk 1 above)
Comment (alfredh) [reply src/dd/dd.c]: Thanks for the review.

I changed the code to `re_printf`

File: src/dd/dd_enc.c (hunk 3 above)
Comment (alfredh) [reply src/dd/dd_enc.c]: I removed the logging instead ...

File: src/dd/putbit.c [hunk 4]
```diff
@@ -62,7 +62,8 @@ int putbit_write(struct putbit *pb, unsigned count, unsigned val)
 
 	for (unsigned i=0; i<count; i++) {
 
-		unsigned bit = (val >> (count-i-1)) & 0x1;
+		unsigned shift = count-i-1;
+		unsigned bit = (val >> shift) & 0x1;
```
Comment (sreimers) [inline src/dd/putbit.c:66]: Looks like the shifting has to be limited (by mask size) to prevent shift count overflow `>=32 (sizeof(unsigned) * 8)`

Comment (sreimers) [reply src/dd/putbit.c:66]: Or maybe a limit check for count is enough

**Pull request merged**